src/fmu/dataio/dataio.py

```diff
@@ -414,15 +414,51 @@ def _get_meta_masterdata(self) -> None:
         logger.info("Metadata for masterdata is set!")
 
     def _get_meta_access(self) -> None:
-        """Get metadata overall (default) from access section in config."""
-        # note that access should be possible to change per object
-        if self._config is None or "access" not in self._config.keys():
-            logger.warning("No access section present")
-            self.metadata4access = None
+        """Get metadata overall (default) from access section in config.
+
+        Access should be possible to change per object, based on user input.
+        This is done through the access_ssdl input argument.
+
+        The "asset" field shall come from the config. This is static information.
+
+        The "ssdl" field can come from the config, or be explicitly given through
+        the "access_ssdl" input argument. If the access_ssdl input argument is present,
+        its contents shall take presedence.
+
+        """
+
+        logger.info("_get_meta_access()")
+
+        if self._config is None:
+            # DISCUSS: Should we allow no config? Or raise when missing?
+            logger.debug("No config is given, returning")
             return
```

Comment:
I think we should not accept. Check for None or {}

Reply:
Suggest to do this in separate PR then, because it is a pretty big refactoring. Not accepting missing config would be inserted much higher in the code. There are tests today that does not provide config, etc.

Reply:
Probably here? https://github.com/equinor/fmu-dataio/blob/1e16fc104bad1d0d157401dabca6aba37b70e8cd/src/fmu/dataio/dataio.py#L390

Reply:
Agree, we postpone